### ISSUE TITLE
reactR/R/dependencies.R: Shinyapps.io deployment fails to load `react-tools.umd.cjs` provided by dependency reactwidget-2.0.0

### ISSUE THREAD

**Issue by bryce-carson** [ashokkrish/episim#148]
# Warning
This warning appears in the JavaScript console.
> Loading failed for the <script> with source “https://bryce-carson.shinyapps.io/Episim/_w_df7e27de/reactwidget-2.0.0/react-tools.umd.cjs”. Episim:51:53

# Background
`reactwidget`, version 2.0.0, is a JavaScript library used internally by [the _reactR_ package](https://github.com/react-R/reactR). _reactR_ depends on <a href="https://irian.to/blogs/what-are-cjs-amd-umd-and-esm-in-javascript/" title="What are UMD and CJS?">reactwidget-2.0.0/react-tools.umd.cjs</a>.

### Is this warning related?
> Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user’s experience. For more help https://xhr.spec.whatwg.org/#sync-warning jquery.min.js:2:81868

I'm unsure it is, because some online commentators point out it does say "deprecated," not "removed." The functionality still exists and should still work. It may not be the cause of the failed load of _react-tools.umd.cjs_.

# Resources
Packer may be an alternative to use reactR, but it would be more ad-hoc. At least it'd be more recently supported. Appsilon's _shiny.react_ would be another choice for integrating `react-chart-editor`.

- https://packer.john-coene.com/
- https://appsilon.github.io/shiny.react/

**Comment by bryce-carson** [comment]
[console-export-2024-7-2_15-27-5.txt](https://github.com/user-attachments/files/16076750/console-export-2024-7-2_15-27-5.txt) contains all of the warnings and errors printed on the JavaScript console when the application is running on Shinyapps.io

**Comment by bryce-carson** [comment]
When the application runs locally, the following message is printed to the JavaScript console.

> The script from “http://127.0.0.1:3171/reactwidget-2.0.0/react-tools.umd.cjs” was loaded even though its MIME type (“application/octet-stream”) is not a valid JavaScript MIME type. 127.0.0.1:3171

**Comment by bryce-carson** [comment]
The following source map errors are also present when running locally.

```
Source map error: request failed with status 404
Resource URL: http://127.0.0.1:3171/core-js-2.5.3/shim.min.js
Source Map URL: shim.min.js.map
Source map error: request failed with status 404
Resource URL: http://127.0.0.1:3171/jquery-3.6.0/jquery.min.js
Source Map URL: jquery.min.js.map
Source map error: request failed with status 404
Resource URL: http://127.0.0.1:3171/shinyWidgets/bootstrap-select-1.14.0-beta2/js/bootstrap-select.min.js
Source Map URL: bootstrap-select.min.js.map
```

**Comment by bryce-carson** [comment]
I have opened an issue with upstream https://github.com/react-R/reactR/issues/86.

**Comment by bryce-carson** [comment]
Downgrading reactR to 0.5.0 solves the issue with no negative impact noticed. 0.6.0 introduces a regression, so for now it is a very good thing we're using renv to lock the version we're using for the project to this working package version.

**Comment by bryce-carson** [comment]
https://github.com/react-R/reactR/issues/86 is being closed soon, as 0.6.1 solves the issue and is being pushed to CRAN today.
